File: src/main/java/com/baofeng/blog/vo/common/Article.java
package com.baofeng.blog.vo.common;

import java.time.LocalDateTime;
import java.util.List;
import lombok.Data;

public class Article {
    
    /**
     * 文章分页响应数据
     */
    @Data
    public static class ArticlePageResponseVO {

        private long total;          // 总记录数
        private int pages;           // 总页数
        private List<ArticleVO> list; // 文章列表
    }

    /**
     * 文章信息（改为 class，MyBatis 需要 setter）
     */
    @Data
    public static class ArticleVO {
        private Long id;
        private String article_title;
        private String article_description;
        private String article_content;
        private Integer view_times;
        private Integer commentsCount;
        private Integer thumbs_up_times;
        private Integer type;
        private String originUrl;
        private LocalDateTime  createdAt;
        private LocalDateTime updatedAt;
        private ArticleStatus status;
        private String article_cover;
        private Boolean is_top;
        private AuthorVO author;
        private List<String> categoryNameList;
        private List<String> tagNameList;
    }
    public enum ArticleStatus {
        DRAFT, PUBLISHED, DELETED;
    }

    /**
     * 作者信息（改为 class，MyBatis 需要 setter）
     */
    @Data
    public static class AuthorVO {
        private Long authorId;
        private String authorName;
        private String authorNickname;
        private String authorAvatar;
    }
}

File: blog-vue3/blog-vue3-front/src/types/blog/article.ts
export interface Article {
  id: string;
  articleCover: string;
  articleTitle: string;
  createdAt: string;
}
export type ArticleListResponse = {
    list: Article[];
    total: number;
}
export interface homeGetArticleListParam {
  pageNum: number;
  pageSize: number;
  keyword?: string;
  sortBy?: string;
  sortOrder?: string;
}

export interface ArticleInfo {
  id: number;
  authorName: string;
  type: number;
  originUrl: string;
  thumbsUpTimes: number;
  authorId: number;
  articleContent: string;
  articleCover?: string;
  articleTitle?: string;
  articleDescription?: string;
  createdAt?: string;
  updatedAt?: string;
  categoryNameList?: string[];
  tagNameList?: string[];
  viewTimes?: number;
  is_top: number;
}
export type ArticleInfoListResponse = {
    total: number,
    list: ArticleInfo[]
} 
export type recommendArticles = {
    previous: ArticleInfo,
    next: ArticleInfo,
    recommend: ArticleInfo[]
}

export interface ArchiveGroup {
  year: number;
  articleList: Article[];
}
export type AritcleInfoLenghtList = {
  length:number,
  
}

export interface ArticleSearch   {
  id: number;
  article_title: string;
  highlight_content: string;
  rest_content: string;
  article_content: string
}
